src/components/SkipLink/SkipLink.ts

```diff
@@ -0,0 +1,71 @@
+export default class SkipLink extends HTMLElement {
+    private $link: HTMLLinkElement | null;
+
+    constructor() {
+        super();
+
+        // Binding
+        this.onClick = this.onClick.bind(this);
+
+        // UI
+        this.$link = this.firstElementChild as HTMLLinkElement;
```

Comment:
I don't know how Astro handles custom elements but with native custom elements, you can't expect an element's children to be available during the `constructor` (unless maybe everything is executed after `DOMContentLoaded`).

Reply:
Web components defined inside of an Astro component are loaded & executed after the main `app.ts` file, which I'm almost certain is executed after `DOMContentLoaded` as you suspected, so it shouldn't break 

That being said, we could indeed move that assignation to the `connectedCallback` function instead just to be safe.